src/Buildsome.hs

```diff
@@ -867,15 +868,22 @@ runCmd bte@BuildTargetEnv{..} entity target = do
   inputs <- readIORef inputsRef
   outputs <- readIORef outputsRef
   builtTargets <- readIORef builtTargetsRef
-  return RunCmdResults
-    { rcrStdOutputs = stdOutputs
-    , rcrSelfTime = realToFrac time
-    , rcrInputs =
+
+  let rcrInputs =
       -- This is because we don't serialize input/output access
       -- (especially when undelayed!) to the same file. So we don't
       -- really have a correct input stat/desc here for such inputs. However, outputs should always considered as mode-only inputs.
-      inputs `M.difference` outputs
-    , rcrOutputs = outputs
+        inputs `M.difference` outputs
+  let rcrOutputs = outputs
+  let shmem = bsSharedMemory bteBuildsome
+  let inputSet = M.keysSet rcrInputs
+  forM_ inputSet $ SharedMemory.sharedMemoryAddFile shmem
```

Comment:
maybe we want to write to shmem also the outputs?
No other command is supposed to generate these outputs, and in case some future job needs them as inputs, they will be ready.